common/src/main/scala/weco/storage_service/bagit/models/MultiChecksumValue.scala

```diff
@@ -0,0 +1,43 @@
+package weco.storage_service.bagit.models
+
+import weco.storage_service.verify.{HashingAlgorithm, MD5, SHA1, SHA256, SHA512}
+
+sealed trait MultiChecksumException extends RuntimeException
+
+object MultiChecksumException {
+  case object NoChecksum extends MultiChecksumException
+  case object OnlyWeakChecksums extends MultiChecksumException
+}
+
+case class MultiChecksumValue[T](
+  md5: Option[T] = None,
+  sha1: Option[T] = None,
+  sha256: Option[T] = None,
+  sha512: Option[T] = None
+) {
+  if (md5.isEmpty && sha1.isEmpty && sha256.isEmpty && sha512.isEmpty) {
+    throw MultiChecksumException.NoChecksum
+  }
+
+  // We support MD5 and SHA1 for backwards compatibility (see RFC 8493 § 2.4)
```

Comment:
I wonder if linking directly to the IETF version of the manifest would be useful (here and elsewhere).